gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApplicationServiceImpl.java

```diff
@@ -524,15 +529,26 @@ private static void validateAndEncodeClientCertificate(
     ) {
         if (applicationSettings.getTls() != null && !StringUtils.isBlank(applicationSettings.getTls().getClientCertificate())) {
             // validate certificate
+            final Certificate[] certificates;
             try {
-                final Certificate[] certificates = KeyStoreUtils.loadPemCertificates(applicationSettings.getTls().getClientCertificate());
-                // For some cases, KeyStoreUtils does not throw an exception but simply returns an empty array of certificates.
-                if (certificates.length == 0) {
-                    throw new InvalidApplicationCertificateException("An error has occurred while parsing client certificate");
-                }
+                certificates = KeyStoreUtils.loadPemCertificates(applicationSettings.getTls().getClientCertificate());
             } catch (Exception e) {
-                throw new InvalidApplicationCertificateException("An error has occurred while parsing client certificate");
+                throw new ApplicationInvalidCertificateException();
+            }
+            // For some cases, KeyStoreUtils does not throw an exception but simply returns an empty array of certificates.
+            if (certificates.length == 0) {
+                throw new ApplicationEmptyCertificateException();
+            }
+            if (certificates.length > 1) {
```

Comment:
Although it is quite unusual to have certificate chain, it could happen (CA+client cert). I would not fail in that case but rather take the last one.
That being I never tried to call a server using mTLS with a cert chain...

Reply:
see [433731c](https://github.com/gravitee-io/gravitee-api-management/pull/8843/commits/433731ca33f553f9b0ea6664ac62bc112cb093e5)